detox/src/DetoxExportWrapper.js

```diff
@@ -35,7 +34,18 @@ class DetoxExportWrapper {
       Detox.none.initContext(global);
     }
 
-    this[_detox] = await DetoxExportWrapper._initializeInstance(config, params);
+    Detox.none.setError(null);
+
+    try {
+      this[_detox] = DetoxExportWrapper._createInstance(config);
+      await this[_detox].init(params);
```

Comment:
This line is super important.
Without assigning the Detox instance to `this[_detox]`, we can't effectively call `detox.cleanup()`, surprise! :boom:
That was a fat bug responsible for why we had issues with artifacts not appearing on test runner timeouts.

Reply:
Fixes https://github.com/wix/Detox/issues/1914?